app/src/main/res/layout/tab_machine.xml

```diff
@@ -14,17 +14,42 @@
         android:id="@+id/addExercise"
         style="@style/Widget.AppCompat.Button.AddButton"
         android:text="Add new Exercise" />
-
-    <AutoCompleteTextView
-        android:id="@+id/searchField"
-        android:layout_width="match_parent"
+    
+    <LinearLayout
+        android:layout_width="fill_parent"
         android:layout_height="wrap_content"
-        android:layout_gravity="center_vertical"
-        android:hint="@string/FilterHint"
-        android:imeOptions="actionNext"
-        android:inputType="text|textCapWords|textAutoComplete|textNoSuggestions|textVisiblePassword"
-        android:nextFocusForward="@+id/SetsInputView"
-        android:singleLine="true" />
+        android:orientation="vertical"
+        android:padding="2dp">
+
+        <RelativeLayout
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content">
+
+            <AutoCompleteTextView
+                android:id="@+id/searchField"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:layout_gravity="center_vertical"
+                android:hint="@string/FilterHint"
+                android:imeOptions="actionNext"
+                android:inputType="text|textCapWords|textAutoComplete|textNoSuggestions|textVisiblePassword"
+                android:nextFocusForward="@+id/SetsInputView"
+                android:singleLine="true" />
+
+            <ImageButton
```

Comment:
The two filtering should be logically linked. In the capture below, the Exercise Type filter is set to Isometrics only but when you start typing on the Text filtering, it shows everything. 
![Capture2](https://user-images.githubusercontent.com/16635534/119227131-bbd17480-bb0c-11eb-9382-82234d49a1ee.PNG)
There should be a logical AND between the two filters.

Reply:
Good point. Solved.